http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.google2.{DiskName, ZoneName}
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.{dummyDate, unmarshalDestroyedDate}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.ExecutionContext
+
+final case class PersistentDiskRecord(id: Long,
+                                      googleProject: GoogleProject,
+                                      zone: ZoneName,
+                                      name: DiskName,
+                                      googleId: Option[GoogleId],
+                                      samResourceId: DiskSamResourceId,
```

Comment:
why do we use this here, but not do the same for GCE runtimes?

Reply:
You mean `DiskSamResourceId`? Runtimes have a field for this but it's called `RuntimeInternalId` which IMO is a bit non-descriptive because it's only used for Sam IDs.

Reply:
I'm wondering, is it reasonable to make a `SamResourceId` data type and use it for all different sam resources

Reply:
I thought about that, was thinking it's good to have type safety to not confuse runtime and disk sam IDs. 

Maybe we could add an ADT like:
```
sealed trait SamResource {
  def resourceId: String
  def resourceType: SamResourceType
}
object SamResource {
  final case class RuntimeSamResource(resourceId: String) {
    override def resourceType: SamResourceType.Runtime
  }
  final case class PersistentDiskSamResource(resourceId: String) {
    override def resourceType: SamResourceType.PersistentDisk
  }
```

Reply:
So I started down this path and it led to a lot of other refactoring… I still think it's valuable but I think I will split it into a separate PR, so this can be merged sooner. Will still apply the other feedback in this PR.

Reply:
Opened https://broadworkbench.atlassian.net/browse/IA-1893